### PR TITLE
Added support for legacy alter table flag

### PR DESCRIPTION
**Pull request by talha970**
Added support for legacy alter table flag
More info [here](https://www.sqlite.org/pragma.html#pragma_legacy_alter_table)

### CODE REVIEW

**Comment by gotson** [comment]
hi, thanks for the PR. Can you run `mvn spotless:apply` and commit the changes ?

File: src/main/java/org/sqlite/SQLiteConfig.java [hunk 1]
```diff
@@ -855,6 +856,19 @@ public void useLegacyFileFormat(boolean use) {
...
+     * command (for changing the name of a table) works as it did in SQLite 3.24.0 (2018-06-04) and
+     * earlier.When the flag is off, using the ALTER TABLE RENAME command will mean that all
+     * references to the table anywhere in the schema will be converted to the new name.
+     *
+     * @param use True to turn on legacy alter table behaviour; false to turn off.
+     * @see <a href="https://www.sqlite.org/pragma.html#pragma_legacy_alter_table</a>
+     */
+    public void setLegacyAlterTableFlag(boolean use) {
+        set(Pragma.LEGACY_ALTER_TABLE, use);
+    }
```
Comment (gotson) [inline src/main/java/org/sqlite/SQLiteConfig.java:870]:
```suggestion
    public void setLegacyAlterTable(boolean flag) {
        set(Pragma.LEGACY_ALTER_TABLE, flag);
    }
```

File: src/main/java/org/sqlite/SQLiteConfig.java [hunk 2]
```diff
@@ -426,6 +426,7 @@ public enum Pragma {
                 "Limit the size of rollback-journal and WAL files left in the file-system after transactions or checkpoints",
                 null),
         LEGACY_FILE_FORMAT("legacy_file_format", "No-op", OnOff),
+        LEGACY_ALTER_TABLE("legacy_alter_table", "Use legacy alter table behavior", OnOff),
```
Comment (gotson) [inline src/main/java/org/sqlite/SQLiteConfig.java:429]: should be placed before `LEGACY_FILE_FORMAT` as to respect alphabetical ordering

**Review by gotson — CHANGES_REQUESTED** [review]

File: src/main/java/org/sqlite/SQLiteConfig.java [hunk 3]
```diff
@@ -862,11 +862,11 @@ public void useLegacyFileFormat(boolean use) {
      * earlier.When the flag is off, using the ALTER TABLE RENAME command will mean that all
      * references to the table anywhere in the schema will be converted to the new name.
      *
-     * @param use True to turn on legacy alter table behaviour; false to turn off.
+     * @param flag True to turn on legacy alter table behaviour; false to turn off.
      * @see <a href="https://www.sqlite.org/pragma.html#pragma_legacy_alter_table</a>
      */
-    public void setLegacyAlterTableFlag(boolean use) {
-        set(Pragma.LEGACY_ALTER_TABLE, use);
+    public void setLegacyAlterTableFlag(boolean flag) {
```
Comment (gotson) [inline src/main/java/org/sqlite/SQLiteConfig.java:868]: can you also change the name from `setLegacyAlterTableFlag` to `setLegacyAlterTable` to be consistent with other methods?

File: src/main/java/org/sqlite/SQLiteDataSource.java [hunk 4]
```diff
@@ -289,11 +289,11 @@ public void setLegacyFileFormat(boolean use) {
      * earlier.When the flag is off, using the ALTER TABLE RENAME command will mean that all
      * references to the table anywhere in the schema will be converted to the new name.
      *
-     * @param use True to turn on legacy alter table behaviour; false to turn off.
+     * @param flag True to turn on legacy alter table behaviour; false to turn off.
      * @see <a href="https://www.sqlite.org/pragma.html#pragma_legacy_alter_table</a>
      */
-    public void setLegacyAlterTableFlag(boolean use) {
-        config.setLegacyAlterTableFlag(use);
+    public void setLegacyAlterTableFlag(boolean flag) {
```
Comment (gotson) [inline src/main/java/org/sqlite/SQLiteDataSource.java:295]: can you also change the name from `setLegacyAlterTableFlag` to `setLegacyAlterTable` to be consistent with other methods?

**Review by gotson — CHANGES_REQUESTED** [review]

File: src/main/java/org/sqlite/SQLiteConfig.java (hunk 3 above)
Comment (talha970) [reply src/main/java/org/sqlite/SQLiteConfig.java]: my bad didn't spot the method rename

**Review by gotson — APPROVED** [review]

**Comment by gotson** [comment]
thanks for your contribution!

**Comment by talha970** [comment]
thank you for reviewing! Can you tell me the expected date of release?

**Comment by gotson** [comment]
> thank you for reviewing! Can you tell me the expected date of release?

no idea, there's a few PRs in flight to review.

The snapshot is already available, you can try that.

**Pull request merged**